### PR TITLE
Add types and functionality needed to start supporting interchain accounts testing

### PR DESCRIPTION
**Pull request by jtieri**
## Description

This PR adds the necessary types and functionality that should be required to support testing interchain accounts.
This PR *does not* add testing scenarios that simulate the usage of interchain accounts.

## Checklist

- [x] I have made sure the upstream branch for this PR is correct
- [x] I have made sure this PR is ready to merge
- [x] I have made sure that I have assigned reviewers related to this project

## Changes

Added methods to the `Chain` interface for registering an interchain account, querying an interchain account, and sending a bank transfer message to an interchain account.

Added methods for creating clients, creating connections, and querying connections to the `Relayer` interface.

Added a test chain that will utilize a docker image for the binary from the interchain accounts [demo](https://github.com/cosmos/interchain-accounts-demo)

Removed newline from end of error message.

### CODE REVIEW

**Review by agouin — APPROVED** [review]
Looks great!

File: chain/cosmos/chain_node.go [hunk 1]
```diff
@@ -861,3 +862,73 @@ func (tn *ChainNode) logger() *zap.Logger {
...
+
+	exitCode, stdout, stderr, err := tn.NodeJob(ctx, command)
+	if err != nil {
+		return "", dockerutil.HandleNodeJobError(exitCode, stdout, stderr, err)
+	}
+
+	// at this point stdout should look like this:
+	// interchain_account_address: cosmos1p76n3mnanllea4d3av0v0e42tjj03cae06xq8fwn9at587rqp23qvxsv0j
+	// we split the string at the : and then just grab the address before returning.
+	parts := strings.SplitN(stdout, ":", 2)
```
Comment (mark-rushakoff) [inline chain/cosmos/chain_node.go:905]: [`strings.Fields`](https://pkg.go.dev/strings#Fields) could be appropriate here too. Is there no `--json` option for output on this command?

File: chain/cosmos/chain_node.go [hunk 2]
```diff
@@ -861,3 +862,73 @@ func (tn *ChainNode) logger() *zap.Logger {
...
+	return strings.TrimSpace(parts[1]), nil
+}
+
+// SendICABankTransfer builds a bank transfer message for a specified address and sends it to the specified
+// interchain account.
+func (tn *ChainNode) SendICABankTransfer(ctx context.Context, connectionID, fromAddr string, amount ibc.WalletAmount) error {
+	msg := fmt.Sprintf(`{
```
Comment (mark-rushakoff) [inline chain/cosmos/chain_node.go:912]: I think producing this JSON is safe here with valid arguments, but I would feel better about it as 
```go
json.Marshal(map[string]interface{}{
  "@type": "/cosmos.bank.v1beta1.MsgSend",
  "from_address": fromAddr,
  "to_address", amount.Address,
  "amount": []map[string]interface{}{
    map[string]interface{}{
      "denom": amount.Denom,
      "amount": amount.Amount.String(), // Or fmt.Sprintf, whatever is appropriate.
    },
  },
})
```

Just to possibility of producing malformed JSON. If this were in wider use I might define local struct for it, but for a one-off, `map[string]interface{}` seems fine.

**Review by mark-rushakoff — APPROVED** [review]

File: chain/cosmos/chain_node.go (hunk 2 above)
Comment (jtieri) [reply chain/cosmos/chain_node.go:912]: Yeah I debated sticking this into a struct but I couldn't think of anywhere else we would use this unless we wanted to support generic bank messages. I'll make the changes so it's at least marshaling and we know its valid JSON.

File: chain/cosmos/chain_node.go (hunk 1 above)
Comment (jtieri) [reply chain/cosmos/chain_node.go:905]: Hmm looking back there is a flag for producing JSON output, must have glossed over it initially.

I'm going to open another PR today that actually adds a test case built on top of this PR so I may actually make this change in that PR.

**Pull request merged**